rules/cross-platform/defense_evasion_masquerading_space_after_filename.toml

```diff
@@ -0,0 +1,60 @@
+[metadata]
+creation_date = "2022/10/18"
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2022/10/18"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rules identifies a process created from an executable with a space appended to the end of the filename. This may
+indicate an attempt to masquerade a malicious file as benign to gain user execution. When a space is added to the end of
+certain files, the OS will execute the file according to it's true filetype instead of it's extension. Adversaries can
+hide a program's true filetype by changing the extension of the file. They can then add a space to the end of the name so
+that the OS automatically executes the file when it's double-clicked.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Masquerading Space After Filename"
+note = """## Setup
+
+If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions <8.2, events will not define `event.ingested` and default fallback for EQL rules was not added until 8.2, so you will need to add a custom pipeline to populate `event.ingested` to @timestamp for this rule to work.
+"""
+references = [
+    "https://www.picussecurity.com/resource/blog/picus-10-critical-mitre-attck-techniques-t1036-masquerading",
+]
+risk_score = 47
+rule_id = "f5fb4598-4f10-11ed-bdc3-0242ac120002"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "macOS", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type == "start" and 
+  (process.args:("* ", "./* ") or process.name: "* ") and not 
```

Comment:
We may want to explore using regex for this instead of a wildcard and space. It should help make the matching more strict. Something similar to the following -> `^[a-zA-Z0-9_.-]+(\s)$`

Reply:
I was wondering about the implications of wildcard vs. regex. I assumed that the wildcard wouldn't miss anything. The image below is why I chose process.name over process.executable originally as I was just trying to match the file name with the space and yes it is still reflected in the name after it starts. I guess the question is, is it best to capture anytime the file is used by another process, including when it's being created with something like "nano". Or is it best to just capture it being executed as the process.executable itself. If that's the case I need to account for it executing from any directory. 

![image](https://user-images.githubusercontent.com/59296946/196744771-512455b7-3c39-4f93-8312-cb8665df32ab.png)

Reply:
Thanks for the explanation! I believe executable would fit properly since process names can easily be altered when spawning and may not accurately reflect the original binary. To account for file paths the regex can be changed to -> `^(\/[a-zA-Z0-9\s_.-]+)+(\s)$`.

This should account for the following:

* Any path in Linux or Mac with alphanumeric characters as well as underscore, hyphens, dots or spaces in them infinite number of times
* Has to start with a path leading character of `/` and end with a space `\s`

Reply:
Still working on a regex expression that will work. We don't support those anchor operators [```^``` and ```$```](https://www.elastic.co/guide/en/elasticsearch/reference/8.4/regexp-syntax.html#regexp-unsupported-operators)

Reply:
@imays11 - Based on your example and some additional testing. I believe `"""/[a-z0-9\s_\-\\./]+\s"""` would work for this matching. Confirmed `\s` can be used to match spaces so it should follow a standard macOS or Linux file path ending with a space. No matches in our telemetry with the space, but removed we have several.

Reply:
@terrancedejesus thank you! I've updated the rule query to use regex instead of wildcards and alert on process.executable instead of process.args and process.name to reduce noise.